python/build.py

```
@@ -289,6 +357,92 @@ def load_binary_tokens(path: str) -> list[int]:
         return list(f.read())
 
 
+def evaluate(gpu_model, bcfg: BuildConfig, val_loader: "BpeDataLoader",
+             max_chunks: int) -> tuple[float, float]:
+    """Run forward-only eval on val set. Returns (avg_loss, perplexity).
+
+    Uses a fresh Conductor per eval (independent CMS state) so eval
+    doesn't corrupt training context. No backward pass, no weight update.
+    """
+    conductor = nl_hecate.Conductor(bcfg.k, bcfg.chunk_sizes)
+    context = nl_hecate.ContextState(bcfg.k, bcfg.d_model)
```

Ruff (0.15.1)

[error] Local variable `context` is assigned to but never used

Remove assignment to unused variable `context`

(F841)